--- wxMac-2.8.12.orig/src/cocoa/app.mm	2011-03-22 13:34:51.000000000 +0100
+++ wxMac-2.8.12/src/cocoa/app.mm	2011-11-04 18:52:28.000000000 +0100
@@ -431,8 +431,8 @@
             subtype:0 data1:0 data2:0] atStart:NO];
 }
 
-extern "C" static void ObserveMainRunLoopBeforeWaiting(CFRunLoopObserverRef observer, CFRunLoopActivity activity, void *info);
-extern "C" static void ObserveMainRunLoopBeforeWaiting(CFRunLoopObserverRef observer, CFRunLoopActivity activity, void *info)
+extern "C" void ObserveMainRunLoopBeforeWaiting(CFRunLoopObserverRef observer, CFRunLoopActivity activity, void *info);
+extern "C" void ObserveMainRunLoopBeforeWaiting(CFRunLoopObserverRef observer, CFRunLoopActivity activity, void *info)
 {
     static_cast<wxApp*>(info)->CF_ObserveMainRunLoopBeforeWaiting(observer, activity);
 }
