diff --git a/include/wx/osx/webview_webkit.h b/include/wx/osx/webview_webkit.h
index 7fea524..db0b562 100644
--- a/include/wx/osx/webview_webkit.h
+++ b/include/wx/osx/webview_webkit.h
@@ -21,6 +21,8 @@
 #include "wx/control.h"
 #include "wx/webview.h"
 
+#include <objc/objc.h>
+
 // ----------------------------------------------------------------------------
 // Web Kit Control
 // ----------------------------------------------------------------------------
@@ -149,7 +151,7 @@ private:
     wxWindowID m_windowID;
     wxString m_pageTitle;
 
-    struct objc_object *m_webView;
+    id m_webView;
 
     // we may use this later to setup our own mouse events,
     // so leave it in for now.
diff --git a/include/wx/osx/webviewhistoryitem_webkit.h b/include/wx/osx/webviewhistoryitem_webkit.h
index 776311c..d556afc 100644
--- a/include/wx/osx/webviewhistoryitem_webkit.h
+++ b/include/wx/osx/webviewhistoryitem_webkit.h
@@ -15,6 +15,8 @@
 #if wxUSE_WEBVIEW && wxUSE_WEBVIEW_WEBKIT && (defined(__WXOSX_COCOA__) \
                                           ||  defined(__WXOSX_CARBON__))
 
+#include <objc/objc.h>
+
 class WXDLLIMPEXP_WEBVIEW wxWebViewHistoryItem
 {
 public:
@@ -27,7 +29,7 @@ public:
 
 private:
     wxString m_url, m_title;
-    struct objc_object *m_histItem;
+    id m_histItem;
 };
 
 #endif // wxUSE_WEBVIEW && wxUSE_WEBVIEW_WEBKIT 
